infogami/infobase/_dbstore/save.py

```diff
@@ -399,15 +399,15 @@ def group_index(self, index):
         """Groups the index based on table.
         """
         groups = defaultdict(dict)
-        for (table, thing_id, property_id), values in index.iteritems():
+        for (table, thing_id, property_id), values in index.items():
             groups[table][thing_id, property_id] = values
         return groups
 
     def ignore_long_values(self, index):
         """The DB schema has a limit of 2048 chars on string values. This function ignores values which are longer than that.
         """
         is_too_long = self._is_too_long
-        return dict((k, [v for v in values if not is_too_long(v)]) for k, values in index.iteritems())
+        return {k: [v for v in values if not is_too_long(v)] for k, values in index.items()}
```

Comment:
Used only once.  Python 2 will build a list in memory which will take up more RAM.  Python 3 will be fast.